### ISSUE TITLE
SC2003 should not warn about "match", i.e. regular expression usage

### ISSUE THREAD

**Issue by rugk** [koalaman/shellcheck#2033]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2003
- My shellcheck version (`shellcheck --version` or "online"): 0.7.0 + online tested
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

The problem here is I got SC2039 first. Now the wiki page for [SC2039](https://github.com/koalaman/shellcheck/wiki/SC2039) could be improved, because it does not actually say what you could do in POSIX here instead.
Actually **it does not mention** regex at all.

However, I've found `expr match` may do the same. (it seems to return the index of the match)

And if it is "antiquated" it surely is POSIX-compatible.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
package=0
if [ "$package" =~ "^package:" ]; then
        echo 1
fi
if expr match  "$package"  "^package:"; then
        echo 1
fi
```

#### Here's what shellcheck currently says:

Line 3:
if [ "$package" =~ "^package:" ]; then
                ^-- SC2039: In POSIX sh, =~ regex matching is undefined.
 
Line 6:
if expr match  "$package" =~ "^package:"; then
   ^-- SC2003: expr is antiquated. Consider rewriting this using $((..)), ${} or [[ ]].

#### Here's what I wanted or expected to see:

Line 3:
if [ "$package" =~ "^package:" ]; then
                ^-- SC2039: In POSIX sh, =~ regex matching is undefined.

**Comment by koalaman** [comment]
`expr match str regex` is not POSIX, but `expr str : regex` is. ShellCheck only recognized the latter and suppressed the warning. ShellCheck now detects this and suggests `expr str : regex` instead, as well as detecting a number of other issues like missing escaping in commands like `expr 2 * 3`, `expr 2 | 3`, and `expr foo : fo*`.

SC2039 was previously retired in favor of one SC3xxx for each individual issue, so the wiki should be more helpful as well.

Thanks!
